packages/aws-cdk/test/integ/run-against-dist.bash

```diff
@@ -155,16 +155,30 @@ function prepare_java_packages() {
 
 function prepare_nuget_packages() {
   # For NuGet, we wrap the "dotnet" CLI command to use local packages.
-  log "Hijacking 'dotnet build' command..."
+  log "Writing new NuGet configuration..."
 
-  ORIGINAL_DOTNET=$(type -p dotnet) || { echo "No 'dotnet' found" >&2; exit 1; }
-  export ORIGINAL_DOTNET
-  export NUGET_SOURCE=$dist_root/dotnet
+  local NUGET_SOURCE=$dist_root/dotnet
 
   if [ ! -d "$NUGET_SOURCE" ]; then
     echo "NuGet packages missing at $NUGET_SOURCE" >&2
     exit 1
   fi
+
+  mkdir -p $HOME/.nuget/NuGet
+  if [ -f $HOME/.nuget/NuGet/NuGet.Config ]; then
+    echo "⚠️ Saving previous NuGet.Config to $HOME/.nuget/NuGet/NuGet.Config.bak"
+    mv $HOME/.nuget/NuGet/NuGet.Config $HOME/.nuget/NuGet/NuGet.Config.bak
```

Comment:
don't you want to restore this?

isn't there a way to use a different location and tell `dotnet` to use that location as their config file instead of the default?

Reply:
Added an EXIT trap to restore.

Unfortunately, not all the `dotnet` commands accept the `--configfile` argument that is used to specify the location of this file, and in particular, `dotnet sln` is used by the dotnet "hooks" and does not support it; yet it will trigger a `restore` operation, which will fail if the configuration isn't proper.